src/empirical.jl

```diff
@@ -2,6 +2,57 @@
 # Licensed under the ISC License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+"Internal function used to calculate the empirical variogram using a spatial index"
+function spatial_index_variogram!(sums, counts, sdata, xi, xj, distance, nlags, Δh, hmax, z₁, z₂, npts)
```

Comment:
Can we shorten the list of arguments a bit? Even if we repeat some code inside each function, this list of argument seems too huge.

Reply:
It's a long list, I agree, though since it's meant as an internal function I think it's ok. We can shorten it only by adding more code duplication, which is also not great.

Do you want a leading underscore before internal function names by the way?